sn_networking/src/cmd.rs

```diff
@@ -348,6 +334,23 @@ impl SwarmDriver {
         Ok(())
     }
 
+    /// Update local peers stored on the SwarmDriver
+    /// Also logs the k-buckets information
+    ///
+    /// This should prevent having to dive into the swarm driver to get the local peers
+    pub(crate) fn update_local_peers(&mut self, changed_peer_id: PeerId) {
+        let mut all_peers: Vec<PeerId> = vec![];
+        for kbucket in self.swarm.behaviour_mut().kademlia.kbuckets() {
```

Comment:
the dive into the kad::kbuckets as I understand shall not taking too much resources and causing noticable mem spike issue.
Actually, this extra container of `self.all_local_peers` could be more to be concerned.

If we do want to reduce the calling to swarm_driver, it will be better put this extra container into node level and send the all_peers info together with the event of PeerAdded or PeerRemoved

Reply:
Right now we do this for every `trigger_replication` call. Which there are many many many of. So I thiiink it can be affecting things. I'm still not sure.

This PR atm suffers from some wild CPU. So it feels liike it's unblocked _something_. Im trying to determine what that is...

Yeh I did wonder about pushing it up to Network. Seemed messier, but it would avoid the call into here and some channel work entirely yeh